pkg/hostpath/hpv.go

```diff
@@ -111,6 +121,11 @@ func (hpv *hostPathVolume) IsAllowed() bool {
 	defer hpv.Lock.Unlock()
 	return hpv.Allowed
 }
+func (hpv *hostPathVolume) IsReadOnly() bool {
+	hpv.Lock.Lock()
+	defer hpv.Lock.Unlock()
+	return hpv.ReadOnly
```

Comment:
So you have to lock the host path volume before you can check if it is ReadOnly? Weird.

Reply:
yep `go test -race` turned this up, both in the unit tests and e2e's (especially the disruptive/restart one), for all the fields of this object

essentially any concurrent activity was flagged

If you are familiar with java, I've essentially created a synchronized java bean